src/dev.rs

```diff
@@ -379,7 +431,8 @@ impl<F: FieldExt> MockProver<F> {
                                     Any::Fixed => {
```

Comment:
(applies to lines 418-420 above)

I didn't know that selectors could be on a different row to the gate's logical row! That seems confusing. Is it necessary at all, or should we enforce that selectors have a row offset of 0?

Reply:
Now that the `Selector` type consistently means a positive boolean selector, _maybe_ rotations are unnecessary for them? I'm fine with restricting this if desired.

Reply:
It is technically possible that there could be optimizations where you share selectors that are on different rows between gates, but the more I think about how confusing that would be, the more I think it should be prohibited. It also potentially interferes with visualizations where you click on a row and see the gates enabled for that row.

(If you *really* want to do those optimizations, you can still do them using fixed columns.)

Reply:
I'll open a separate PR for this.

Reply:
https://github.com/zcash/halo2/pull/287